File: scripts/convert_json_to_excel.py
import json
import pandas as pd
from pathlib import Path

# Paths
json_path = r'C:\Users\C-R\Desktop\Asper Beauty Box\Asper Beauty shop prodcuts\product apify\dataset_productss_2026-01-15_23-29-45-343.json'
output_path = r'C:\Users\C-R\Desktop\Asper Beauty Box\lovable\products_for_upload.xlsx'

print("Loading JSON file...")
with open(json_path, 'r', encoding='utf-8') as f:
    products = json.load(f)

print(f"Found {len(products)} products")

# Convert to format expected by bulk upload UI
# Based on BulkUpload.tsx, it expects: sku, name, costPrice, sellingPrice
excel_data = []

for product in products:
    # Extract price from variants
    current_price = 0
    cost_price = 0
    
    if product.get('variants') and len(product['variants']) > 0:
        variant = product['variants'][0]
        if variant.get('price', {}).get('current'):
            current_price = float(variant['price']['current']) / 100  # Convert from cents
        if variant.get('price', {}).get('previous'):
            cost_price = float(variant['price']['previous']) / 100
    
    # Use source ID as SKU
    sku = product.get('source', {}).get('id', '')
    if not sku:
        sku = f"PROD-{len(excel_data) + 1}"
    
    excel_data.append({
        'SKU': str(sku),
        'Product Name': product.get('title', 'Untitled Product'),
        'Cost Price': cost_price if cost_price > 0 else current_price * 0.7,  # Estimate 30% margin if no cost
        'Selling Price': current_price if current_price > 0 else 10.0  # Default price
    })

# Create DataFrame
df = pd.DataFrame(excel_data)

# Save to Excel
print(f"Saving to {output_path}...")
df.to_excel(output_path, index=False, sheet_name='Products')

print(f"✅ Successfully converted {len(excel_data)} products to Excel!")
print(f"📁 File saved to: {output_path}")
print(f"\nColumn mapping:")
print(f"  - SKU: Product identifier")
print(f"  - Product Name: Full product title")
print(f"  - Cost Price: Cost/previous price")
print(f"  - Selling Price: Current selling price")

File: scripts/convert_json_to_excel.mjs
import fs from "fs";
import ExcelJS from "exceljs";

const jsonPath =
  "C:/Users/C-R/Desktop/Asper Beauty Box/Asper Beauty shop prodcuts/product apify/dataset_productss_2026-01-15_23-29-45-343.json";
const outputPath =
  "C:/Users/C-R/Desktop/Asper Beauty Box/lovable/products_for_upload.xlsx";

console.log("Loading JSON file...");
const rawData = fs.readFileSync(jsonPath, "utf8");
const products = JSON.parse(rawData);

console.log(`Found ${products.length} products`);

// Create workbook
const workbook = new ExcelJS.Workbook();
const worksheet = workbook.addWorksheet("Products");

// Add headers
worksheet.columns = [
  { header: "SKU", key: "sku", width: 20 },
  { header: "Product Name", key: "name", width: 50 },
  { header: "Cost Price", key: "costPrice", width: 15 },
  { header: "Selling Price", key: "sellingPrice", width: 15 },
];

// Convert products
let count = 0;
for (const product of products) {
  // Extract price from variants
  let currentPrice = 0;
  let costPrice = 0;

  if (product.variants && product.variants.length > 0) {
    const variant = product.variants[0];
    if (variant.price?.current) {
      currentPrice = parseFloat(variant.price.current) / 100; // Convert from cents
    }
    if (variant.price?.previous) {
      costPrice = parseFloat(variant.price.previous) / 100;
    }
  }

  // Use source ID as SKU
  const sku = product.source?.id?.toString() || `PROD-${count + 1}`;

  worksheet.addRow({
    sku: sku,
    name: product.title || "Untitled Product",
    costPrice: costPrice > 0 ? costPrice : currentPrice * 0.7, // Estimate 30% margin
    sellingPrice: currentPrice > 0 ? currentPrice : 10.0, // Default price
  });

  count++;
}

// Save to Excel
console.log(`Saving to ${outputPath}...`);
await workbook.xlsx.writeFile(outputPath);

console.log(`✅ Successfully converted ${count} products to Excel!`);
console.log(`📁 File saved to: ${outputPath}`);
console.log(`\nColumn mapping:`);
console.log(`  - SKU: Product identifier`);
console.log(`  - Product Name: Full product title`);
console.log(`  - Cost Price: Cost/previous price`);
console.log(`  - Selling Price: Current selling price`);
